tests/unit/s2n_tls_prf_test.c

```diff
@@ -310,5 +312,63 @@ int main(int argc, char **argv)
         };
     };
 
+    /* s2n_prf tests */
```

Comment:
Should we also test that the new PRF is producing correct results? We can assume that awslc has tested it, but we have some extra logic around seeds that we could be messing up. Maybe we should at least test that the two different methods produce the same result?

Reply:
There are known-value tests in s2n_tls_prf_test for both the [TLS 1.2](https://github.com/aws/s2n-tls/blob/b9c4d60de49193cea31f0acfbddc47cc66d41d2c/tests/unit/s2n_tls_prf_test.c#L75-L84) and [pre-TLS 1.2 PRF](https://github.com/aws/s2n-tls/blob/b9c4d60de49193cea31f0acfbddc47cc66d41d2c/tests/unit/s2n_tls_prf_test.c#L39-L46) versions. Additionally, a known-value test that tests the three-seed version of this function exists in [s2n_tls_hybrid_prf_test](https://github.com/aws/s2n-tls/blob/b9c4d60de49193cea31f0acfbddc47cc66d41d2c/tests/unit/s2n_tls_hybrid_prf_test.c#L109-L110). Are there any other tests you think would be worth adding to the unit tests?

We discussed offline about adding a fuzz test that compares both implementations, and I will add this in a future PR: https://github.com/aws/s2n-tls/issues/4031